app/models/order_balance.rb

```diff
@@ -1,21 +1,21 @@
 # frozen_string_literal: true
 
 class OrderBalance
-  delegate :zero?, :abs, :to_s, to: :to_f
+  delegate :zero?, :abs, :to_s, :to_f, :to_d, :<, :>, to: :amount
```

Comment:
I'd like to try if using Ruby's [Comparable](https://ruby-doc.org/core-3.0.0/Comparable.html) and implementing the `<=>` saves us from having to implement each method separately. Could you take a look @Matt-Yorkley ?